crux-llvm/src/Crux/LLVM/Config.hs

```diff
@@ -82,6 +89,7 @@ data LLVMOptions = LLVMOptions
   , noCompile :: Bool
   , optLevel :: Int
   , symFSRoot :: Maybe FilePath
+  , supplyMainArguments :: Bool
```

Comment:
Do we think that we might want more expressive arguments here in the future? If we made this an ADT, we would have an easy path in the future for just adding a new constructor:
```
data MainArguments = NoArguments -- ^ main ()
                                   | EmptyArguments  -- ^ main (int argc, char* argv[]) where argc = 0
                                   | SomeArguments (NonEmpty String) -- ^ main(int argc, char* argv[]) where argc > 0
```

Reply:
You're right—I could foresee someone wanting `SomeArguments` in the future. I can easily turn the `yes`/`no` options for `supply-main-arguments` to `empty`/`none` to keep the door open for extending it in the future. Does `config-schema` allow parsing `empty`, `none`, and something like `["args", "go", "here"]` all in the same option? My understanding of `config-schema`'s [`ValueSpec`](https://hackage.haskell.org/package/config-schema-1.2.2.0/docs/Config-Schema-Spec.html#t:ValueSpec) data type is that it permits this, but I haven't tried implementing something like that yet.

Reply:
Ah, I'm actually not sure what config-schema supports. I don't see anything quite like that in the exposed API, but that doesn't mean one could definitely not write something like that. No need to do this right now, given that we don't know when we'll need that capability.

Given the structure of `config-schema`, we might instead need to specify command line arguments by supplying multiple `--argument` parameters and just saying that specifying that is mutually-exclusive with `supply-main-arguments`.

Reply:
I prototyped what this might look like with the following code:

```hs
supplyMainArgumentsSpec :: ValueSpec SupplyMainArguments
supplyMainArgumentsSpec =
  (NoArguments <$ atomSpec "none") <!>
  (EmptyArguments <$ atomSpec "empty") <!>
  (SomeArguments <$> Crux.listSpec Crux.stringSpec)
```

To my delight, this works exactly like I expected it would. In light of this, I'm quite happy with this design, since it should be straightforward to add `SomeArguments` later if we desire it.